taca/utils/undetermined.py

```diff
@@ -0,0 +1,164 @@
+    quality thresholds are met.
+    """
+    if os.path.exists(os.path.join(run, dmux_folder)):
+        xtp=cl.XTenParser(run)
+        ss=xtp.samplesheet
+        lb=xtp.lanebarcodes
+        path_per_lane=get_path_per_lane(run, ss)
+        barcode_per_lane=get_barcode_per_lane(ss)
+        workable_lanes=get_workable_lanes(run, status)
+        for lane in workable_lanes:
+            if is_unpooled_lane(ss,lane):
+               if check_index_freq(run,lane, freq_tresh):
+                    if first_qc_check(lane,lb, und_tresh, q30_tresh):
+                        link_undet_to_sample(run, lane, path_per_lane)
+            else:
+                logger.warn("The lane {}  has been multiplexed, according to the samplesheet and will be skipped.".format(lane))
+
+    else:
+        logger.warn("No demultiplexing folder found, aborting")
+
+def get_workable_lanes(run, status):
+    """List the lanes that have a .fastq file"""
+    lanes=[]
+    pattern=re.compile('L00([0-9])')
+    for unde in glob.glob(os.path.join(run, dmux_folder, 'Undetermined_*')):
+        name=os.path.basename(unde)
+        lanes.append(int(pattern.search(name).group(1)))
+    lanes=list(set(lanes))
+    if status =='IN_PROGRESS': 
+        #the last lane is the one that is currently being worked on by bcl2fastq, don't work on it.
+        lanes=lanes[:-1]
+    logger.info("going to work with lanes {}".format(lanes))
+    return lanes
+
+
+def link_undet_to_sample(run,lane, path_per_lane):
+    """symlinks the undetermined file to the right sample folder"""
+    for fastqfile in glob.glob(os.path.join(run, dmux_folder, 'Undetermined_*_L00{}_*'.format(lane))):
+        logger.info("linking file {} to {}".format(fastqfile, path_per_lane[lane]))
+        os.symlink(fastqfile, os.path.join(path_per_lane[lane], os.path.basename(fastqfile)))
+
+def save_index_count(barcodes, run, lane):
+    """writes the barcode counts"""
+    with open(os.path.join(run, dmux_folder, 'index_count_L{}.tsv'.format(lane)), 'w') as f:
+        for barcode in sorted(barcodes, key=barcodes.get, reverse=True):
+            f.write("{}\t{}\n".format(barcode, barcodes[barcode]))
+
+def check_index_freq(run, lane,freq_tresh):
+    """uses subprocess to perform zcat <file> | sed -n '1~4 p' | awk -F ':' '{print $NF}', counts the barcodes and 
+    returns true if the most represented index accounts for less than freq_tresh% of the total"""
+    barcodes={}
+    if os.path.exists(os.path.join(run, dmux_folder,'index_count_L{}.tsv'.format(lane))):
+        logger.info("Found index count for lane {}, skipping.".format(lane))
+        return False
+    else:
+        open(os.path.join(run, dmux_folder,'index_count_L{}.tsv'.format(lane)), 'a').close()
+        for fastqfile in glob.glob(os.path.join(run, dmux_folder, 'Undetermined_*_L00{}_R1*'.format(lane))):
+            logger.info("working on {}".format(fastqfile))
+            zcat=subprocess.Popen(['zcat', fastqfile], stdout=subprocess.PIPE)
```

Comment:
It can be done in python, Hugo had some great ideas, but it was a bit more clunky than this.